app/src/main/java/com/lightricks/feedexercise/ui/feed/FeedViewModel.kt

```diff
@@ -33,8 +36,35 @@ open class FeedViewModel : ViewModel() {
         refresh()
     }
 
+    @SuppressLint("CheckResult")
     fun refresh() {
-        //todo: fix the implementation
+        NetworkAccess.feedApiService.getFeed()
+            .subscribeOn(Schedulers.io())
+            .observeOn(AndroidSchedulers.mainThread())
+            .subscribe({ feedResponse ->
+                handleResponse(feedResponse)
+            },{ error ->
+                handleNetworkError(error)
+            })
+    }
+
+    fun handleResponse(feedResponse: MetadataItemsList) {
+        convertFeedResponseToFeedItemList(feedResponse)
+    }
+
+    fun handleNetworkError(error: Throwable) {
+        Log.e("Test: ", error.message.toString())
+    }
+
+    fun convertFeedResponseToFeedItemList(metadataItemsList: MetadataItemsList) {
+        val tempFeedList = ArrayList<FeedItem>()
+        metadataItemsList.templatesMetadata.forEach { metadataItem ->
```

Comment:
Use `map` instead for clarity and performance. (map knows what the size of the target list it has to instantiate).